File: codes/week15/지원/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        si = 0
        sl = len(s)

        if sl == 0:
            return True

        for tc in t:
            if s[si] == tc:
                si += 1

                if si == sl:
                    return True

        return False

File: codes/week15/진현/392.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int tg = 0, idx = 0;
        while (tg < s.size() && idx < t.size()) {
            while (idx < s.size() && s[tg] != t[idx])
                idx++;
            
            if (idx < s.size()) {
                tg++;
                idx++;
            }
            else {
                break;
            }
        }
            
        return tg == s.size();
    }
};